### ISSUE TITLE
Rejected incoming transaction not signaled on broacast_tx_sync response

### ISSUE THREAD

**Issue by nicsor** [tendermint/tendermint#8928]
**Tendermint version**:
v0.35.7

**ABCI app**:
kvstore

**Environment**:
- **OS** (e.g. from /etc/os-release): Debian 11 with docker image: tendermint/tendermint:v0.35.7

**What happened**:
After spamming with data on `broacast_tx_sync`, and filling the mempool we receive no indication that the transaction was rejected through the response.

**What you expected to happen**:
in the response to the `broacast_tx_sync` request I expect 'code' to be different than 0 as it was on the versions prior to v0.35. for example, on v0.34.14. I understand the field 'mempool_error' will be deprecated in future releases, but it might be useful to know that the transaction was rejected, at least through `code`

**Have you tried the latest version**: yes

**How to reproduce it** (as minimally and precisely as possible):
Spam with transaction until mempool is full, checking the total on http://127.0.0.1:26667/num_unconfirmed_txs
If necessary, decrease mempool size or throttle cpu resources of docker image

**Logs (paste a small part showing an error (< 10 lines) or link a pastebin, gist, etc. containing more of the log file)**:
```
ERROR rejected incoming good transaction; mempool full err="mempool is full: number of txs 5000 (max: 5000), total txs bytes 1827884 (max: 1073741824)" module=mempool tx=E4B1564A0C9B4A0B99289C22F84B8394F4C6B84F7EDFA8C5BB66587FD56FD26F version=v1
```
the response on v0.35.* is typically:
```
{
  "jsonrpc": "2.0",
  "id": -1,
  "result": {
    "code": 0,
    "data": "",
    "log": "",
    "codespace": "",
    "mempool_error": "",
    "hash": "C07FEA24938EF8052712AB3D8B238B2337315105D375CFD75079919C0213B59B"
  }
}
```

**Config (you can paste only the changes you've made)**:
  default settings with 2 or 4 validators

**Comment by cmwaters** [comment]
Thanks @nicsor for opening the issue. I think we missed adding the error to `mempool_error` when returning. I will check to see if an error is returned for v0.36

**Comment by nicsor** [comment]
Thank you for the update.
Unfortunately, the above changes do not fix the issue on v0.35. Issue still exists in v0.35.9.

I'm assuming I should be looking at the v0.35.x branch.
Basically the above changes have updated: addNewTransaction, but after that, the return is still nil.

https://github.com/tendermint/tendermint/blob/b4eaccd242b875511dc758751fe375348300ebbf/internal/mempool/v1/mempool.go#L241

Which seems in tune with this comment: https://github.com/tendermint/tendermint/blob/b4eaccd242b875511dc758751fe375348300ebbf/internal/mempool/v1/mempool.go#L172

**Comment by nicsor** [comment]
This change  https://github.com/tendermint/tendermint/commit/27c523dccbe7f2f577868ee871b0042c1fef9a53
is on branch 0.35x, while the docker image I think is based is on branch v0.35x

very confusing branch naming
